test/e2e/httpproxy/default_global_rate_limiting_test.go

```diff
@@ -0,0 +1,368 @@
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo",
+								Port: 80,
+							},
+						},
+					},
+				},
+			},
+		}
+		p, _ = f.CreateHTTPProxyAndWaitFor(p, e2e.HTTPProxyValid)
+
+		// Wait until we get a 200 from the proxy confirming
+		// the pods are up and serving traffic.
+		res, ok := f.HTTP.SecureRequestUntil(&e2e.HTTPSRequestOpts{
+			Host:      p.Spec.VirtualHost.Fqdn,
+			Condition: e2e.HasStatusCode(200),
+			RequestOpts: []func(*http.Request){
+				e2e.OptSetHeaders(map[string]string{
+					"X-Default-Header": "test_value_5",
+				}),
+			},
+		})
+		require.NotNil(t, res, "request never succeeded")
+		require.Truef(t, ok, "expected 200 response code, got %d", res.StatusCode)
+
+		// Make another request against the proxy, confirm a 429 response
+		// is now gotten since we've exceeded the rate limit.
+		res, ok = f.HTTP.SecureRequestUntil(&e2e.HTTPSRequestOpts{
+			Host:      p.Spec.VirtualHost.Fqdn,
+			Condition: e2e.HasStatusCode(200),
+			RequestOpts: []func(*http.Request){
+				e2e.OptSetHeaders(map[string]string{
+					"X-Default-Header": "test_value_5",
+				}),
+			},
+		})
+		require.NotNil(t, res, "request never succeeded")
+		require.Truef(t, ok, "expected 200 response code, got %d", res.StatusCode)
+	})
+
+	Specify("default global rate limit policy is set but HTTPProxy has its own global rate limit policy", func() {
+		t := f.T()
+
+		f.Fixtures.Echo.Deploy(namespace, "echo")
+		f.Certs.CreateSelfSignedCert(namespace, "echo-cert", "echo", "globalratelimitroutetls.projectcontour.io")
+
+		p := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "defaultglobalratelimitvhostnontls",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				VirtualHost: &contourv1.VirtualHost{
+					Fqdn: "defaultglobalratelimitvhostnontls.projectcontour.io",
+					TLS: &contourv1.TLS{
+						SecretName: "echo",
+					},
+					RateLimitPolicy: &contourv1.RateLimitPolicy{
+						Global: &contourv1.GlobalRateLimitPolicy{
+							Descriptors: []contourv1.RateLimitDescriptor{
+								{
+									Entries: []contourv1.RateLimitDescriptorEntry{
+										{
+											GenericKey: &contourv1.GenericKeyDescriptor{
+												Value: "foo",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Routes: []contourv1.Route{
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo",
+								Port: 80,
+							},
+						},
+					},
+				},
+			},
+		}
+		p, _ = f.CreateHTTPProxyAndWaitFor(p, e2e.HTTPProxyValid)
+
+		// Make another request against the proxy, confirm a 429 response
+		// is now gotten since we've exceeded the rate limit.
+		res, ok := f.HTTP.SecureRequestUntil(&e2e.HTTPSRequestOpts{
+			Host:      p.Spec.VirtualHost.Fqdn,
+			Condition: e2e.HasStatusCode(429),
+			RequestOpts: []func(*http.Request){
+				e2e.OptSetHeaders(map[string]string{
+					"X-Default-Header": "test_value_6",
```

Comment:
same comment as above on sending the header here

Reply:
might want to add another unique descriptor for this test if we want to ensure the descriptor set on the httpproxy causes the rate limit to be triggered